org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/YAMLLanguageServer.java

```diff
@@ -21,19 +21,53 @@
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+import org.eclipse.lsp4e.LanguageServersRegistry;
+import org.eclipse.lsp4e.LanguageServersRegistry.LanguageServerDefinition;
+import org.eclipse.lsp4e.LanguageServiceAccessor;
 import org.eclipse.lsp4e.server.ProcessStreamConnectionProvider;
 import org.eclipse.lsp4j.DidChangeConfigurationParams;
+import org.eclipse.lsp4j.InitializeResult;
 import org.eclipse.lsp4j.jsonrpc.messages.Message;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage;
 import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.wildwebdeveloper.Activator;
 import org.eclipse.wildwebdeveloper.embedder.node.NodeJSManager;
 
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 
+@SuppressWarnings("restriction")
 public class YAMLLanguageServer extends ProcessStreamConnectionProvider {
+	private static final String SETTINGS_KEY = "settings";
+	private static final String YAML_KEY = "yaml";
+	private static final String VALIDATE_KEY = "validate";
+	private static final String COMPLETION_KEY = "completion";
+	private static final String HOVER_KEY = "hover";
+	private static final String SCHEMAS_KEY = "schemas";
 	
-	private String cachedSchema;
+ 	private static final IPreferenceStore store = Activator.getDefault().getPreferenceStore();
+ 	private static final LanguageServerDefinition yamlLsDefinition = LanguageServersRegistry.getInstance().getDefinition("org.eclipse.wildwebdeveloper.yaml");
+ 	private static final IPropertyChangeListener psListener = new IPropertyChangeListener() {
```

Comment:
Why would we need to change those preferences to false? I suggest we simply get rid of the preference listener, and hardcode the values to yes, until someone has a good reason to disable it.

Reply:
That's not about enabling or disabling schemas processing. It's about the loading schema itself. Listener updates the schema on server when a user changed it YAML Schemas preference page. So if a schemas preference is not empty (user defined or changed some schemas) we're sending the new schemas to the server and enabling it to work with schemas.
So, we need this listener.
The only point missed here, probably, is that YAML Preference page has no schemas defined, it should disable schemas processing on the server. (No schemas loaded - no processing is needed)

Reply:
OK, got it.

Reply:
I was wrong on these preferences. Independently on existence of schemsa  these preferences are to be set tu true as yaml file still needs to be validated. This was the reason of failing `TestLanguageServers.testYAMLFile()` test case.